src/internal_module/fs.rs

```diff
@@ -0,0 +1,862 @@
+use crate::event_loop::PollResult;
+use crate::quickjs_sys::*;
+use std::convert::TryInto;
+use std::fs;
+use std::fs::Permissions;
+use std::io;
+use std::os::wasi::fs::{FileTypeExt, MetadataExt};

```

Comment:
this feature need nigthly rust, but your can implement it use `std::os::wasi::prelude::FromRawFd` and `wasi_snapshot_preview1.fd_filestat_get`